include/seqan/sequence/string_set_base.h

```diff
@@ -1784,11 +1784,13 @@ inline SEQAN_FUNC_ENABLE_IF(And<Is<ContainerConcept<TSequences> >,
 write(TTarget & target, TSequences const & seqs, TDelim const & delimiter)
 {
     typedef typename Size<TSequences const>::Type TSize;
+    typedef typename Iterator<TSequences const>::Type TSourceIt;
+
     if (SEQAN_UNLIKELY(empty(seqs)))
         return;
-    for (TSize i = 0; i < length(seqs) - 1; ++i)
+    for (TSourceIt it = begin(seqs), itBack = (end(seqs) - 1); it != itBack; ++it)
     {
```

Comment:
Nice catch! But I experienced a lot of troubles if you are not exact with the Iterator type. Maybe add the Standard() tag to begin and end so it won't mess up.